ofrak_core/ofrak/gui/server.py

```diff
@@ -410,39 +458,75 @@ async def queue_patch(self, request: Request) -> Response:
         start = int(start_param) if start_param is not None else 0
         end_param = request.query.get("end")
         end = int(end_param) if end_param is not None else (await resource.get_data_length())
-
+        # TODO: There has to be a better way
+        new_data_string = "\\x" + "\\x".join(
+            [new_data.hex()[i : i + 2] for i in range(0, len(new_data.hex()), 2)]
+        )
+        script_str = (
+            r"""
+        {resource}.queue_patch"""
+            rf"""(Range({start}, {end}), b"{new_data_string}")"""
+        )
+        """await {resource}.save()"""
```

Comment:
Is this string supposed to be dead code?

```suggestion
```

Reply:
This should actually be included in the script so that the queued patch is saved into the script, no? I had accidentally left this out when adjusting the script string resource name replacements from `$resource` to `{resource}`.

Unfortunately, as yet another example of the fragility of how the script builder handles whitespace for scripts, there's no elegant way to add this onto the existing script string in this route handler, which is using auto-concatenation to build the `queue_patch()` call. Black isn't able to parse the input script if I simply include variations of `"""\nawait {resource}.save()"""` to the existing string. In order to handle thing, I need to create and add a second action within this route handler, which doesn't seem ideal. It works though.

```
        script_str = (
            r"""
        {resource}.queue_patch"""
            rf"""(Range({start}, {end}), b"{new_data_string}")"""
        )
        await self.script_builder.add_action(resource, script_str, ActionType.MOD)
        script_str = """
        await {resource}.save()"""
        await self.script_builder.add_action(resource, script_str, ActionType.MOD)
```